### PR TITLE
Support patch operations with both id and value

### PR DESCRIPTION
**Pull request by michaelkantor**

### CODE REVIEW

File: test/SpecRunner.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE HTML>
```
Comment (nilgradisnik) [inline test/SpecRunner.html]: Why is this file here?

Comment (michaelkantor) [inline test/SpecRunner.html]: Copy/paste
